src/main/java/com/amplitude/api/PinnedAmplitudeClient.java

```diff
@@ -286,11 +322,24 @@ protected SSLSocketFactory getPinnedCertSslSocketFactory(SSLContextBuilder conte
         if (sslSocketFactory == null) {
             try {
                 sslSocketFactory = context.build().getSocketFactory();
-                logger.i(TAG, "Pinning SSL session using Comodo CA Cert");
+                if (context.serverZone == AmplitudeServerZone.EU) {
+                    logger.i(TAG, "Pinning SSL session using AWS Root CA Cert");
+                } else {
+                    logger.i(TAG, "Pinning SSL session using Comodo CA Cert");
+                }
             } catch (Exception e) {
                 logger.e(TAG, e.getMessage(), e);
             }
         }
         return sslSocketFactory;
     }
+
+    @Override
+    public AmplitudeClient setServerZone(AmplitudeServerZone serverZone) {
+        super.setServerZone(serverZone);
+        this.initializedSSLSocketFactory = false;
+        this.sslSocketFactory = null;
+        this.initialize(this.context, this.apiKey, this.userId);
```

Comment:
For cal initialize, would there be any other thing affected? Like config, user id, device id, etc.? Also if there is no api key, would it be fine?

Reply:
It should be fine.
1. in iOS, we saved the userId and deviceId in the DB. If we can get the value from the DB, we will not re-initialize it. Other configurations are not overwritten in the initialize step. 
2. It will be fine. It will be the same error as the first initialization.